### PR TITLE
Ensure, that `Report` will always be `Send + Sync + 'static`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

It's required, that `Report` is always `Send + Sync + 'static` for using it in multi-threading. This is a bug introduced in #774 

## 🔍 What does this change?

- Add a compile-time test for ensuring it's `Send`, `Sync`, and `'static`
- Fix this by adding `Send` + `Sync` bounds on `FrameImpl`

### CODE REVIEW

File: packages/libs/error-stack/src/lib.rs [hunk 1]
```diff
@@ -454,3 +454,19 @@ pub use self::{
...
+    fn test_send<T: Send>() {}
+    fn test_sync<T: Sync>() {}
+    fn test_static<T: 'static>() {}
+
+    fn report() {
+        test_send::<Report<()>>();
+        test_sync::<Report<()>>();
+        test_static::<Report<()>>();
+    }
+}
```
Comment (Alfred-Mountfield) [inline packages/libs/error-stack/src/lib.rs:472]:
```suggestion
}

```

File: packages/libs/error-stack/src/lib.rs [hunk 2]
```diff
@@ -454,3 +454,19 @@ pub use self::{
     macros::*,
     report::Report,
 };
+
+
```
Comment (TimDiekmann) [inline packages/libs/error-stack/src/lib.rs:458]:
```suggestion
```

**Review by Alfred-Mountfield — APPROVED** [review]
Do we want to consider some multi-threaded integration tests? I assume if we had some then it would have picked this up.

Not sure the value in the unit-tests as is.

**Comment by TimDiekmann** [comment]
I don't think it's worth doing separated tests for this. Having these tests in place covers the topic well enough.

**Comment by Alfred-Mountfield** [comment]
To be clear I'm not suggesting multi-threaded tests just to ensure we don't accidentally remove bounds, but more to make sure we continue to have the correct bounds and such to work nicely across threads. If supporting usage in multi-threaded applications is a core part of our library, I think we should test it. Not necessarily as part of this PR though

**Review by Alfred-Mountfield — APPROVED** [review]
📎

**Pull request merged**